### PR TITLE
Issue #11067: Remove powermock workaround for JDK16

### LINKED ISSUE #11067: Remove `illegal-access=warn` workaround for JDK16
From discussion in #10947, we can remove this workaround for powermock issue:

https://github.com/checkstyle/checkstyle/blob/e2fa4a1d1e5de4625b4db77cdd4804aa5b13ddbb/pom.xml#L1893-L1902

### PR DESCRIPTION
**Pull request by nmancus1**
Closes #11067

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1890,17 +1890,6 @@
 
   <profiles>
 
-    <!-- Till https://github.com/powermock/powermock/issues/1094 -->
```
Comment (rnveach) [inline pom.xml:1893]: It looks like new issue was not necessary.

**Review by rnveach — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (rnveach) [reply pom.xml:1893]: Please confirm if this issue can be closed when this is merged.

Comment (romani) [reply pom.xml:1893]: it is issue not in our repo.
I think it is ok to keep #11067 .

**Review by romani — APPROVED** [review]
ok to merge if CI pass

File: pom.xml (hunk 1 above)
Comment (rnveach) [reply pom.xml:1893]: My bad. I saw issue and did not look at it closer. Resolving chain.

**Pull request merged**